lib/stages/asdfFileWrite.cpp

```diff
@@ -0,0 +1,182 @@
+#include <Stage.hpp>
+#include <StageFactory.hpp>
+#include <asdf/asdf.hxx>
+#include <atomic>
+#include <cassert>
+#include <chordMetadata.hpp>
+#include <cstdint>
+#include <errno.h>
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <memory>
+#include <optional>
+#include <prometheusMetrics.hpp>
+#include <sstream>
+#include <string>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <utility>
+#include <vector>
+#include <visUtil.hpp>
+
+namespace {
+ASDF::scalar_type_id_t chord2asdf(const chordDataType type) {
+    switch (type) {
+        case int4p4:
+            return ASDF::id_uint8; // TODO: Define ASDF int4+4 type
+        case int8:
+            return ASDF::id_int8;
+        case int16:
+            return ASDF::id_int16;
+        case int32:
+            return ASDF::id_int32;
+        case int64:
+            return ASDF::id_int64;
+        case float16:
+            return ASDF::id_float16;
+        case float32:
+            return ASDF::id_float32;
+        case float64:
+            return ASDF::id_float64;
+        default:
+            assert(0);
+    }
+}
+} // namespace
+
+class asdfFileWrite : public kotekan::Stage {
```

Comment:
We should comment this, I think it's most or less just a copy of the `hdf5FileWrite` stage comments.

Reply:
I more-or-less copied in the HDF5 documentation :)